sorc/reg2grb2.fd/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+list(APPEND fortran_src
+reg2grb2.f
+regdiag.f
+)
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -free -convert big_endian -fp-model precise -assume byterecl")
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^(GNU)$")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fconvert=big-endian")
+endif()
+
+set(exe_name reg2grb2.x)
+add_executable(${exe_name} ${fortran_src})
+target_link_libraries(
+  ${exe_name}
+  landsfcutil::landsfcutil_d
+  ip::ip_d
+  sp::sp_d
+  bacio::bacio_4
+  w3nco::w3nco_4
```

Comment:
has anyone verified that all of these are actually used in this program?

Reply:
I went through the makefile and compared which ones were being linked. I didn't go through the code line by line.

Reply:
That's a question for @BrianCurtis-NOAA, but landsfcutil is a new dependency for this PR, so I assume yes for that at least.

Reply:
@aerorahul the old build_reg2grb2.sh has landsfcutil in a LIBS export which was then used in the makefile that was used to build reg2grb2. I don't see any subroutines that seem to be using any landsfcutil subroutines, but if needed I will spend more time on going line by line to see if i missed something. Let me know.